framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -896,31 +910,63 @@ protected String dotOutputFileName(UnderlyingAST ast) {
         return null;
     }
 
+    /**
+     * Returns the type of the left-hand side of an assignment without
+     * applying local variable defaults to type variables.
+     *
+     * The type variables that are types of local variables are defaulted to
```

Comment:
We can add a new "LOCAL_VARIABLE_DEFAULT_FOR_TYPE_INFERENCE" location :-)
Anyways, let's open an issue to not forget about this.